File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>
#include <cstring>
using namespace std;
vector<vector<int>> v;
int answer = 0;
int vis[10001];
void bfs(int start){
    queue<int> q;
    q.push(start);
    vis[start] = true;
    
    while(!q.empty()){
        int current = q.front();
        q.pop();
        
        for(int i=0; i<v.size(); i++) {
            if(i == current) continue;
            if(!vis[i] && v[current][i] == 1) {
                vis[i] = true;
                q.push(i);
            }
        }
    }
}
int solution(int n, vector<vector<int>> computers) {
    v = computers;
    memset(vis, 0, sizeof vis);
    for(int i = 0; i < v.size(); i++){
        if(!vis[i]){
            bfs(i);
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
let vis = [];
let q = [];

function bfs(start, coms){
    q.push(start);
    vis[start] = true;
    
    while(q.length > 0){
        let cur = q[0];
        q.shift();
        
        for(let i = 0; i < coms.length; i++){
            if(!vis[i] && coms[cur][i] == 1){
                q.push(i);
                vis[i] = true;
            }
        }
    }
}

function solution(n, computers) {
    var answer = 0;
    vis = Array(n).fill(0);
    for(let i = 0; i < computers.length; i++){
        if(!vis[i]){
            bfs(i, computers);
            answer++;
        }
    }
    return answer;
}